types/src/block_metadata.rs

```diff
@@ -38,19 +45,39 @@ impl BlockMetaData {
         }
     }
 
-    pub fn into_inner(
-        self,
-    ) -> (
-        HashValue,
-        u64,
-        BTreeMap<AccountAddress, Ed25519Signature>,
-        AccountAddress,
-    ) {
-        (
-            self.id,
-            self.timestamp_usec,
-            self.previous_block_votes,
-            self.proposer,
-        )
+    pub fn into_inner(self) -> Result<(ByteArray, u64, ByteArray, AccountAddress)> {
```

Comment:
This seems potentially error-prone (e.g., what if the caller mixes up the id bytearray and the votes bytearray?). In addition, it seems like only the VM will want to convert this into VM-only types like `ByteArray`. Perhaps it would make sense to put this logic there instead?

Reply:
I did thought about the issue before but the thing is we can't implement the logic at VM side unless we make the field public, which is something I'm not so sure of.